source_code/code_parsing/code_handle.py

```diff
@@ -0,0 +1,65 @@
+from typing import Dict, Iterable, List
+
+from dulwich.repo import Repo
+from joblib import delayed, parallel_backend, Parallel
+from tqdm import tqdm
+
+from .repo_parser import RepoParser
+from source_code.git_repo_extract.repo_ops import operate_temporary_repo
+
+
+def parallelize_extraction(temp_repo_path: str,
+                           parsed_lines: Iterable,
+                           supported_languages: List[str],
+                           commits_limit: int):
+    """
+    Method decomposes extract_from_json function
+
+    :param commits_limit: maximum number of commits that should be parsed inside of repo
+    :param temp_repo_path: path to folder, where temporaryDirectories for repositories should be created
+    :param parsed_lines: lines of different repos commit_info data
+    :param supported_languages: which programming languages should be overviewed
+    :return:
+    """
+    with parallel_backend(backend="multiprocessing"):
+        parallel = Parallel(n_jobs=1, verbose=100)
```

Comment:
hardcoded value - make it an argument
(optional, but it's better to fix - it's quite strong antipatttern)